Sources/FINNFilterImplementation/FINNFilterSelectionData.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import Foundation
+
+class FINNFilterSelectionData: NSObject {
+    private struct GeoKey {
+        static let latitude = "lat"
+        static let longitude = "lon"
+        static let radius = "radius"
+        static let locationName = "geoLocationName"
+    }
+
+    private(set) var selectionValues: [String: [String]]
+
+    init(selectionValues: [String: [String]]) {
+        self.selectionValues = selectionValues
+    }
+
+    convenience override init() {
+        self.init(selectionValues: [:])
+    }
+}
+
+extension FINNFilterSelectionData {
+    func setSelectionValues(_ values: [String], for key: String) {
+        selectionValues[key] = values
+    }
+
+    func setSelectionValue(_ value: String, for key: String) {
+        setSelectionValues([value], for: key)
+    }
+
+    func addSelectionValue(_ value: String, for key: String) {
+        var values: [String]
+        if let previousValues = selectionValues[key] {
+            values = previousValues
+        } else {
+            values = []
+        }
+        values.append(value)
+        setSelectionValues(values, for: key)
+    }
+
+    func removeSelectionValue(_ value: String, for key: String) {
+        if let previousValues = selectionValues[key] {
+            setSelectionValues(previousValues.filter({ $0 != value }), for: key)
+        }
+    }
+
+    func removeSelectionValues(_ key: String) {
+        selectionValues.removeValue(forKey: key)
+    }
+
+    func selectionValues(for name: String) -> [String] {
```

Comment:
Should it be `key` instead of `name`? Just because it's called `key` in other methods.